core/common/hexutil.cpp

```diff
@@ -29,17 +42,21 @@ namespace kagome::common {
     return hex_lower(bytes.data(), bytes.size());
   }
 
-  expected::Result<std::vector<uint8_t>, std::string> unhex(
-      std::string_view hex) {
+  outcome::result<std::vector<uint8_t>> unhex(std::string_view hex) {
     std::vector<uint8_t> blob((hex.size() + 1) / 2);
 
-    boost::format
-        error_format;  // for storing formatted error message if any occurs
     try {
       boost::algorithm::unhex(hex.begin(), hex.end(), blob.begin());
```

Comment:
check if unhex has last argument ```boost::system::error_code& ec```.

If so, then you don't need to try/catch.

just 
```
boost::system::error_code ec;
boost::algorithm::unhex(hex.begin(), hex.end(), blob.begin(), ec);
if(ec){
  return ec;
}
return blob;
```

Reply:
I didn't find overloads with this sort of argument.